docs/Modern-GraphQLInRelay.md

```diff
@@ -142,7 +142,25 @@ Will cause a generated file to appear in `./__generated__/MyComponent.graphql`,
 with both runtime artifacts (which help to read and write from the Relay Store)
 and [Flow types](https://flow.org/) to help you write type-safe code.
 
-The Relay Compiler is responsible for generating code as part of a build step which, at runtime, can be used statically. By building the query ahead of time, the client's JS runtime is not responsible for generating a query string, and fields that are duplicated in the query can be merged during the build step, to improve parsing efficiency. If you have the ability to persist queries to your server, the compiler's code generation process provides a convenient time to convert a query or mutation's text into a unique identifier, which can greatly reduce the upload bytes required in some applications.
+The Relay Compiler is responsible for generating code as part of a build step which, at runtime, can be used statically. By building the query ahead of time, the client's JS runtime is not responsible for generating a query string, and fields that are duplicated in the query can be merged during the build step, to improve parsing efficiency. 
+
+### Persisting queries
+The Relay Compiler can convert a query or mutation's text into a unique identifier during compilation. This can greatly reduce the upload bytes required in some applications.
+Using a unique identifier in place of your query or mutation also allows you to whitelist operations that are allowed on your server which improves security.
```

Comment:
Should this expand a little more on _how_ it may improve security so people can decide if that applies to them? (Possibly by linking to a good article on query depth as a possible DOS vector.)

Reply:
There's two parts to that:

* Security against internal DOS
* Security against full schema introspection (which may or may not be undesirable)

Reply:
Good point and done.